jack.py

```diff
@@ -739,12 +739,21 @@ def set_process_callback(self, callback, userdata=None):
             is called.
 
         """
-        @self._callback("JackProcessCallback", error=STOP_CALLING)
-        def callback_wrapper(frames, _):
-            return callback(frames, userdata)
+        if isinstance(callback, _ffi.CData):
+            if _ffi.typeof(callback) != _ffi.typeof("JackProcessCallback"):
+                raise ValueError("Expected %s, got %s" % \
+                                 (_ffi.typeof("JackProcessCallback"),
+                                  _ffi.typeof(callback)))
```

Comment:
This is too much error handling code for my taste.
You should check for both conditions in the same `if` expression and let the interpreter throw an error later if the type is wrong (in line 752).